### PR TITLE
Bug 1789560: Various fixes for newest SDK version and API compatibility

### PR DESCRIPTION
**Pull request by zvonkok**
Various fixes for newest SDK version and API compatibility

### CODE REVIEW

File: assets/master/0100_service_account.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: nfd-master
-  namespace: openshift-nfd
+#  namespace: openshift-nfd
```
Comment (sjug) [inline assets/master/0100_service_account.yaml]: Why are these comments?

File: pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go [hunk 2]
```diff
@@ -147,18 +131,13 @@ func ClusterRoleBinding(n NFD) (ResourceStatus, error) {
...
 	obj := n.resources[state].ClusterRoleBinding
 
-	if len(n.ins.Spec.OperandNamespace) != 0 {
-		obj.SetNamespace(n.ins.Spec.OperandNamespace)
-	}
-
 	found := &rbacv1.ClusterRoleBinding{}
 	logger := log.WithValues("ClusterRoleBinding", obj.Name, "Namespace", obj.Namespace)
 
-	if err := controllerutil.SetControllerReference(n.ins, &obj, n.rec.scheme); err != nil {
```
Comment (sjug) [inline pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go]: How did you decide which objects needed references?

File: pkg/controller/nodefeaturediscovery/nodefeaturediscovery_state.go [hunk 3]
```diff
@@ -38,9 +37,7 @@ func (n *NFD) init(r *ReconcileNodeFeatureDiscovery,
 	n.ins = i
 	n.idx = 0
 
-	secv1.AddToScheme(r.scheme)
-
-	addState(n, "/opt/nfd/config")
+	//addState(n, "/opt/nfd/config")
```
Comment (sjug) [inline pkg/controller/nodefeaturediscovery/nodefeaturediscovery_state.go]: Do we need this?

File: Makefile [hunk 4]
```diff
@@ -1,12 +1,12 @@
 REGISTRY       ?= quay.io
 ORG            ?= zvonkok
-TAG            ?= latest #$(shell git rev-parse --short HEAD)
+TAG            ?= $(shell git branch | grep \* | cut -d ' ' -f2)
```
Comment (sjug) [inline Makefile]: awk life
```
git branch | awk '/\*/ { print $2 }'
```

**Comment by sjug** [comment]
Please clean up the duplicate commits

File: pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go (hunk 2 above)
Comment (zvonkok) [reply pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go]: Namespaced resource cannot be a controllerreference for cluster-wide resources.

File: assets/master/0100_service_account.yaml (hunk 1 above)
Comment (zvonkok) [reply assets/master/0100_service_account.yaml]: I will remove it now, the tests are green now.

File: pkg/controller/nodefeaturediscovery/nodefeaturediscovery_state.go (hunk 3 above)
Comment (zvonkok) [reply pkg/controller/nodefeaturediscovery/nodefeaturediscovery_state.go]: No we do not need this, will remove it.

**Comment by zvonkok** [comment]
@sejug will squash and merge

**Comment by sjug** [comment]
Please cleanup your 13 duplicate commits before merge.

**Comment by zvonkok** [comment]
@sjug the merge will be a "Squash and Merge"

**Comment by sjug** [comment]
@zvonkok What happened to squashing? You merged in 22 commits to master in a single PR...

**Comment by zvonkok** [comment]
@sjug d*** it, next time I need to verify that my squash really worked, sorry will take more care next time!

**Pull request merged**